core/src/test/scala/org/apache/spark/benchmark/BenchmarkBase.scala

```diff
@@ -83,6 +85,11 @@ abstract class BenchmarkBase {
 
   def suffix: String = ""
 
+  /**
+   * Any code before running any benchmark, e.g., data preparation
+   */
+  def beforeAll(): Unit = {}
```

Comment:
My suggestion is not to add the `beforeAll()` method in the current pr If we find it to be widely applicable, we can add this method in a subsequent pr and also extract `beforeAll()` for more microbenchmark scenarios.